Misc/NEWS.d/next/Core and Builtins/2024-03-09-11-10-53.gh-issue-112087.nbI0Pw.rst

```diff
@@ -0,0 +1 @@
+:class:`list` is now compatible with the implementation of :pep:`703`.
```

Comment:
This PR should be the final PR for https://github.com/python/cpython/issues/112087.
I think that we can close the PR, and after GIL is disabled, we can track relevant issues from a separate issue.

Reply:
One more thing I noticed here is that `list.sort` should have a critical section